cmake/cuda.cmake

```diff
@@ -6,6 +6,9 @@ if (${CMAKE_VERSION} VERSION_GREATER_EQUAL "3.18.0")
     cmake_policy(SET CMP0104 OLD)
 endif ()
 
+# use shared
+set(CMAKE_CUDA_RUNTIME_LIBRARY Shared)
```

Comment:
Why do we introduce `CMAKE_CUDA_RUNTIME_LIBRARY`?
This is brought in since cmake 3.17, but we claim that the minimum version requirement of cmake is 3.14

Reply:
Can we make this MSVC only?